core/src/Cuda/Kokkos_Cuda_Team.hpp

```diff
@@ -782,7 +782,7 @@ parallel_scan(const Impl::ThreadVectorRangeBoundariesStruct<
         // exclusive scan -- the final accumulation
         // of i's val will be included in the second
         // closure call later.
-        if (i < loop_boundaries.end && threadIdx.x > 0) {
+        if (i - 1 < loop_boundaries.end && threadIdx.x > 0) {
```

Comment:
this missing -1 was an existing thing in develop right? 

https://github.com/kokkos/kokkos/blob/2e743674a89dc133a6b16e0bc7aa992713e0dbc9/core/src/Cuda/Kokkos_Cuda_Team.hpp#L810

Reply:
This comes from 4a819b6b799b07e818727d49650571a02d87d5d1.

Reply:
yes i mean in develop was missing -1

Reply:
@cz4rs which existing bug? is there an issue linked?

Reply:
I don't think we had an issue open for it.

Reply:
I guess this did not show up because of some gaps in the test suite?

Reply:
Yes, but we are fixing this gap in this pull request (or rather the test we try to enable for `Cuda` as well). 🙂

Reply:
Agree!